plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -0,0 +1,122 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaultserviceplan
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated/internalversion"
+	internalversion "github.com/kubernetes-incubator/service-catalog/pkg/client/listers_generated/servicecatalog/internalversion"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+const (
+	// PluginName is name of admission plug-in
+	PluginName = "DefaultServicePlan"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(io.Reader) (admission.Interface, error) {
+		return NewDefaultServicePlan()
+	})
+}
+
+// exists is an implementation of admission.Interface.
+// It checks to see if Service Instance is being created without
+// a Service Plan if there is only one Service Plan for the
+// specified Service and defaults to that value.
+// that the cluster actually has support for it.
+type defaultServicePlan struct {
+	*admission.Handler
+	scLister internalversion.ServiceClassLister
+}
+
+var _ = scadmission.WantsInternalServiceCatalogInformerFactory(&defaultServicePlan{})
+
+func (d *defaultServicePlan) Admit(a admission.Attributes) error {
+	// we need to wait for our caches to warm
+	if !d.WaitForReady() {
+		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
+	}
+
+	// We only care about service Instances
+	if a.GetResource().Group != servicecatalog.GroupName || a.GetResource().GroupResource() != servicecatalog.Resource("instances") {
+		return nil
+	}
+	instance, ok := a.GetObject().(*servicecatalog.Instance)
+	if !ok {
+		return errors.NewBadRequest("Resource was marked with kind Instance but was unable to be converted")
+	}
+	// If the plan is specified, let it through and have the controller
+	// deal with finding the right plan, etc.
+	if len(instance.Spec.PlanName) > 0 {
+		return nil
+	}
+
+	sc, err := d.scLister.Get(instance.Spec.ServiceClassName)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return errors.NewInternalError(err)
+		}
+		msg := fmt.Sprintf("ServiceClass %q does not exist, can not figure out the default Service Plan.", instance.Spec.ServiceClassName)
+		glog.V(4).Info(msg)
+		return admission.NewForbidden(a, fmt.Errorf(msg))
```

Comment:
that's really weird because I know I've seen some complaint from some golang checking tool complain about using things like Printf() w/o any variable args.  Maybe it was just a dream... :-)